src/components/MessageList/utils/getReadStates.js

```diff
@@ -0,0 +1,32 @@
+export const getReadStates = (messages, read) => {
+  // create object with empty array for each message id
+  const readData = {};
+
+  for (const message of messages) {
+    // Filter out date seperators
+    if (!message || !message.id) {
+      continue;
+    }
+    readData[message.id] = [];
+  }
+
+  for (const readState of Object.values(read)) {
```

Comment:
might be helpful to put a comment on this for loop describing what's happening for future reference